Metadata checker

adamrp opened the pull request:
Adds an additional check to see if metadata can be retrieved for a particular sample.  Note that this touches python code, so the web server will have to be restarted after deployment.

Good barcodes you might use to test:
000001002 (should be all good)
000005739 (this is an animal sample, so metadata pulldown fails)
010001002 (this sample does not exist)
000001055 (this sample is missing sample data)

Commit:
Add proc to check metadata for a single barcode

Commit:
Add function to call new procedure

Commit:
Add check for metadata pulldown

adamrp commented:
@ElDeveloper and @teravest, can this get tested ASAP?  Now that geocoding of all samples has finished, it would be awesome to get this merged so that we can move forward with AG round 7 (those plating the samples will be able to use this to ensure that nothing is sequenced for which we can't successfully pull down the metadata, which has been a problem in the past that has theoretically been fixed now).

teravest commented:
I will test it today.

On Wed, Feb 5, 2014 at 9:57 AM, adamrp notifications@github.com wrote:

> @ElDeveloper https://github.com/ElDeveloper and @teravesthttps://github.com/teravest,
> can this get tested ASAP? Now that geocoding of all samples has finished,
> it would be awesome to get this merged so that we can move forward with AG
> round 7 (those plating the samples will be able to use this to ensure that
> nothing is sequenced for which we can't successfully pull down the
> metadata, which has been a problem in the past that has theoretically been
> fixed now).
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/610#issuecomment-34209224
> .

adamrp commented:
Cool, thanks!

On Wed, Feb 5, 2014 at 9:59 AM, teravest notifications@github.com wrote:

> I will test it today.
> 
> On Wed, Feb 5, 2014 at 9:57 AM, adamrp notifications@github.com wrote:
> 
> > @ElDeveloper https://github.com/ElDeveloper and @teravest<
> > https://github.com/teravest>,
> > can this get tested ASAP? Now that geocoding of all samples has finished,
> > it would be awesome to get this merged so that we can move forward with
> > AG
> > round 7 (those plating the samples will be able to use this to ensure
> > that
> > nothing is sequenced for which we can't successfully pull down the
> > metadata, which has been a problem in the past that has theoretically
> > been
> > fixed now).
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHub<
> > https://github.com/qiime/qiime_web_app/pull/610#issuecomment-34209224>
> > .
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/610#issuecomment-34209848
> .

Commit:
Remove commented sections, add proc exec example

Commit:
Remove email type from MD pulldown fail

Commit:
Change div_ids to be more specific

Commit:
Change no_metadata color to a different color

adamrp commented:
I think I have addressed all the comments so far, so unless there are more, this should be ready to go!

teravest commented:
Looks good to me

ElDeveloper commented on www/wwwadmin/barcode_util.psp:
```diff
@@ -137,9 +135,20 @@ try:
+                # should never get here (this would happen if the metadata
+                # pulldown returned more than one row for a single barcode)
+                div_id = "md_pulldown_error"
+                message = "Please notify Adam that you saw this message..."
```
Could you change this message to describe the problem and add suggest contacting one of the DB admins.

Commit:
Change error message

ElDeveloper commented:
From what we discussed during our meeting today and to have a record of this:
- [ ] Remove the query that does this on bulk for a set of barcodes. This includes removing this query from the repo, `git rm` is your friend.
- [ ] Add interface on python to replace the other query.
- [ ] Make a simple script under `python_code/scritps/`.

Commit:
Add code to call stored procedure, change names

Also removes the query that we had been using for metadata pulldowns.
From now on, metadata pulldowns can be done with python_code/ag_get_metadata.py

adamrp commented:
Okay, how does this look?  Tested a little bit, seems to work fine.

ElDeveloper commented:
@adamrp Would you mind moving `ag_get_metadata.py` to live under `python_code/scripts/`?

ElDeveloper commented:
This looks just like what we want.

adamrp commented:
Will do!
On Feb 7, 2014 4:48 PM, "Yoshiki Vázquez Baeza" notifications@github.com
wrote:

> @adamrp https://github.com/adamrp Would you mind moving
> ag_get_metadata.py to live under python_code/scripts/?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/610#issuecomment-34519855
> .

ElDeveloper commented:
Cool, thanks!
On Feb 7, 2014, at 5:19 PM, adamrp notifications@github.com wrote:

> Will do! 
> On Feb 7, 2014 4:48 PM, "Yoshiki Vázquez Baeza" notifications@github.com 
> wrote: 
> 
> > @adamrp https://github.com/adamrp Would you mind moving 
> > ag_get_metadata.py to live under python_code/scripts/? 
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/610#issuecomment-34519855 
> > . 
> > 
> > —
> > Reply to this email directly or view it on GitHub.

Commit:
Move script to python_scripts/scripts

Commit:
Add QIIME interface to ag_get_metadata.py

Commit:
Merge branch 'metadata_checker' of github.com:adamrp/qiime_web_app into metadata_checker

Commit:
Fill in QIIME script interface, add doc string

adamrp commented:
Okay, this should be ready for merge (perhaps after a re-review of the new changes)

ElDeveloper commented:
Re-reviewing right now, thanks for making the changes!

On Feb 10, 2014, at 11:57 AM, adamrp notifications@github.com wrote:

> Okay, this should be ready for merge (perhaps after a re-review of the new changes)
> 
> —
> Reply to this email directly or view it on GitHub.

ElDeveloper commented:
:+1:
